### ISSUE TITLE
Cannot install from source or release on Firefox 87.0

### ISSUE THREAD

**Issue by vwheezy22** [puemos/hls-downloader-web-extension#178]
**Describe the bug**
Because Mozilla removed this add-on from their page #134 , I am forced to install manually. At first, I tried to install the latest release (the first zip file). Knowing that zip files and xpi files are the exact same (correct me if I'm wrong), I renamed it and chose to "Install Add-On From File" and then "Load Temporary Add-on" from Mozilla's Add-on debug page. I received this error: ```extension-archive.xpi does not contain a valid manifest```

**To Reproduce**
1. After installing the latest release zip file or after building with `./build.sh`, go to "about.addons".
2. Click on the gear in the top right.
3. Select "Debug Add-ons."
4. Select "Load Temporary Add-on" and load the extension archive.

**Expected behavior**
For it to work. I'm not sure about the general structure of add-ons but a brief inspection of `manifest.json` doesn't immediately hint at any invalidity.

**Desktop (please complete the following information):**
 - OS: Linux 5.11.10-arch1-1
 - Browser: Firefox
 - Version 87.0

**Additional context**
I see that this code base is in need of a "Installation Instructions" section. I'd be more than happy to write something up to help me and others in the future to install this great add-on. Resources/quick-start guides on extension development may be helpful as well. Thanks!

**Comment by mjmatthiesen** [comment]
Would be great with @puemos could comment on this. The way I see it, this is FOSS, so he doesn't owe us anything, but ti would be great if there was an update regarding the potential of getting this back on Firefox.

**Comment by puemos** [comment]
Hey, sorry for the silence... my day work is consuming all of my coding time :/
The extension had some issues with the storage technology of Firefox, i will try to take another look at it.
Thanks for the support 😊

**Comment by vwheezy22** [comment]
No worries, if you want to point me to some resources I could try to write-up the installation instructions and fix the possible issue. I'd love to contribute in any way that I can.

**Comment by salmonmolon** [comment]
@vwheezy22 I can install it fine, just move the files from dist to the root of the folder and compress again.

**Comment by vwheezy22** [comment]
@salmonmolon Sorry for the late response.

Tested out your approach just now and it kind of works. My problem is that it only installs in FF's `Debug Mode` which doesn't allow the extension to be persistent. Installing on FF regularly from the `Install Add-on from File` option gives me a corrupt add-on error. I created my `.xpi` file with your method from the release download since I'm getting errors from yarn when trying to build.

Upon further testing from the extension loaded in debug mode, it successfully downloads the `.ts` file but FF says that the `Download Failed`. Pressing `Retry Download` immediately errors out.
